x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -180,19 +180,31 @@ public void execute(EsqlQueryRequest request, EsqlExecutionInfo executionInfo, P
         analyzedPlan(parsed, executionInfo, request.filter(), new EsqlCCSUtils.CssPartialErrorsActionListener(executionInfo, listener) {
             @Override
             public void onResponse(LogicalPlan analyzedPlan) {
-                SubscribableListener.<LogicalPlan>newForked(l -> preOptimizedPlan(analyzedPlan, l))
-                    .<LogicalPlan>andThen((l, p) -> preMapper.preMapper(optimizedPlan(p), l))
-                    .<Result>andThen((l, p) -> executeOptimizedPlan(request, executionInfo, planRunner, p, l))
-                    .addListener(listener);
+                optimizeAndExecute(request, executionInfo, planRunner, analyzedPlan, listener);
             }
         });
     }
 
+    // visible for testing in CsvTests
+    public void optimizeAndExecute(
```

Comment:
I wish this could be protected or default visibility, however CsvTests are in different package.

I wonder if in future it would be possible to extend EsqlSession in CsvTests, override `analyzedPlan` with custom test resolution and reduce visibility of this method.

Reply:
The tests can make a `TestEsqlSession` inside of the same package. Right?